godotproject/Menus/HighScoreScreen/HighScoreScreen.gd

```diff
@@ -0,0 +1,238 @@
+		# (In case we updated it and didn't persist it before).
+		# We need to do this now because this will properly deal with the
+		# "current_level_data".
+		_save_to_disk()
+	current_level = name
+	if name in all_level_data:
+		current_level_data = all_level_data[name]
+	else:
+		# We have to create a new set of level data for this level.
+		current_level_data = []
+		for i in range(5):
+			# Make some fake high scores
+			var stars = max(0, 2-i)
+			stars *= stars
+			stars = min(3, stars)
+			var time = 30*i+30
+			var score = _calculate_score(stars,time)
+			current_level_data.append({'stars':stars, 'time':time, 'score':score})
+	level_label.text = name
+	_display_score_data(current_level_data)
+
+func _display_score_data(score_data, user_i = -1):
+	# Remove previously-existing labels from the HighScoresContainer
+	for i in range(hsc.get_child_count()-1, 3, -1):
+		if hsc.get_child_count() > i:
+			var child = hsc.get_child(i)
+			if child == null:
+				continue
+			child.queue_free()
+	# Always make 6 rows of labels, but don't put the #6
+	# text in the 6th label, and if we don't have enough score data,
+	# don't put any text in the labels in the last row.
+	for i in range(6):
+		var font = sub_font
+		var color = sub_color
+		# Show the user's score in a different colour/font
+		if i == user_i:
+			font = sub_fancy_font
+			color = sub_fancy_color
+
+		# Number
+		var num_lb := Label.new()
+		num_lb.align = Label.ALIGN_CENTER
+		num_lb.add_font_override("font", font)
+		num_lb.add_color_override("font_color", color)
+		if i != 5:
+			num_lb.text = "#%d" % [i+1]
+		hsc.add_child(num_lb)
+		# Stars
+		var star_lb := Label.new()
+		star_lb.align = Label.ALIGN_CENTER
+		star_lb.add_font_override("font", font)
+		star_lb.add_color_override("font_color", color)
+		if i != 5 or len(score_data) >= 6:
+			star_lb.text = "×%d" % score_data[i]['stars']
+		hsc.add_child(star_lb)
+		# Time
+		var time_lb := Label.new()
+		time_lb.align = Label.ALIGN_CENTER
+		time_lb.add_font_override("font", font)
+		time_lb.add_color_override("font_color", color)
+		if i != 5 or len(score_data) >= 6:
+			time_lb.text = Global.format_time(score_data[i]['time'])
+		hsc.add_child(time_lb)
+		# Score
+		var score_lb := Label.new()
+		score_lb.align = Label.ALIGN_CENTER
+		score_lb.add_font_override("font", font)
+		score_lb.add_color_override("font_color", color)
+		if i != 5 or len(score_data) >= 6:
+			score_lb.text = "%06d" % score_data[i]['score']
+		hsc.add_child(score_lb)
+
+func _ready():
+	# Load "all_level_data" from disk.
+	_load_from_disk()
+
+func _process(_delta):
+	self.visible = (current_state != State.HIDDEN)
+	if current_state == State.HIDDEN:
+		pass
+	elif current_state == State.WAIT_FOR_PRESS:
+		if Input.is_action_pressed("jump") \
+		or Input.is_action_pressed("menu") \
+		or Input.is_action_pressed("tongue"):
+			Global.fade_to_scene(load_scene_after_high_score)
+
+# Saving to disk is defeatured for Demo 2
```

Comment:
excellent, I was going to ask if you've planned for when we need store this in non-volatile memory. Gold star!